src/futures/timer.rs

```diff
@@ -0,0 +1,101 @@
+//! Timers
+
+use core::future::Future;
+
+/// A count down timer
+///
+/// # Contract
+///
+/// - `self.start(count); block!(self.wait());` MUST block for AT LEAST the time specified by
```

Comment:
The whole documentation here needs to be adapted to using futures, including examples.